hail/src/main/scala/is/hail/utils/richUtils/RichString.scala

```diff
@@ -8,4 +8,14 @@ class RichString(val str: String) extends AnyVal {
 
     def strings: (String, String) = (truncate, str)
   }
+
+  def equalsCI(other: String): Boolean =
```

Comment:
I'm also pretty sure this will be slower than calling `lower()` on the string before entering the parseSpecialNum method. This doesn't avoid an allocation -- scala will allocate a range object here and call `foreach` on it.